server/src/http-exception.filter.ts

```diff
@@ -1,5 +1,27 @@
 import { ExceptionFilter, Catch, ArgumentsHost, HttpException } from '@nestjs/common';
-import { Request, Response } from 'express';
+import { Response } from 'express';
+
+// response objects from HttpExceptions may be recursively
+// nested to reflect a stack trace (each level is a layer
+// in the stack). This flattens them out so that each response
+// is a top-level item in the errors array.
+export function unfoldedStackTrace(response, status) {
+  const firstError = {
```

Comment:
Teensy nit, but I don't know if "firstError" was the right name — when it recurses, it'll be the second, third, however deep the nesting goes.